### DISCUSSION TITLE
Use of RAUC environnement variable in Handler

### DISCUSSION THREAD

**Discussion by hatmaalla** [rauc/rauc#1016] [Q&A]
Hello,
I use RAUC 1.8 in my project.
I add a call to a handler in my local rauc configuration file under /etc/rauc/system.conf.
In my handler script, I try to use the 'version' value of the manifest file by calling the 'RAUC_MF_VERSION' environment variable.
unfortunately, the content of RAUC_MF_VERSION is always empty !
Does the environement variable created by RAUC can be used within a handler?
Nb:
in the RAUC documentation, I found the following information
https://rauc.readthedocs.io/en/latest/reference.html?highlight=manifest#manifest
````
A free version field that can be used to provide and track version information. No checks will be performed on this version by RAUC itself, although a manager can use this information to reject updates.

````

**Comment by JSydll** [comment]
The short answer is probably: no. The environments exposed to the system-side handlers is different from the one used for the slot-hooks. See [this part of the documentation](https://rauc.readthedocs.io/en/latest/reference.html?highlight=manifest#custom-handlers-interface) on details which env vars are available to the handlers. The variable you search for is only set for hooks in the bundle.

We had a similar situation in our project and ended up with manually parsing the bundle manifest and getting the version from there. It can be done in two, three lines of bash so we are happy with this approach. And you have `RAUC_BUNDLE_MOUNT_POINT` available so you don't even need to fiddle around much with paths.

**Reply by hatmaalla** [reply]
Hi @JSydll 
Thank you for your answer, it seems a good workarround for me ;)
It's better to update the documentation by replacing **handler** by **hook** in the following section to avoid confusion
 "...although a handler can use this information to reject updates."
https://rauc.readthedocs.io/en/latest/reference.html?highlight=manifest#manifest
Thank you again for your support

**Reply by JSydll** [reply]
I mean, the handler _can_ use the information from the bundle manifest - so the documentation is not really wrong 😉 Maybe a clarifying statement below this section could help to raise awareness that not everything is available out of the box in every scenario, though...
